kernel/src/syscall/stat.rs

```diff
@@ -77,6 +81,60 @@ pub fn sys_fstatat(
     Ok(SyscallReturn::Return(0))
 }
 
+pub fn sys_statx(
+    dirfd: FileDesc,
+    filename_ptr: Vaddr,
+    flags: u32,
+    mask: u32,
+    statx_buf_ptr: Vaddr,
+    ctx: &Context,
+) -> Result<SyscallReturn> {
+    let user_space = ctx.user_space();
+    let filename = user_space.read_cstring(filename_ptr, MAX_FILENAME_LEN)?;
+    let flags =
+        StatxFlags::from_bits(flags).ok_or(Error::with_message(Errno::EINVAL, "invalid flags"))?;
+    let mask = StatxMask::from_bits_truncate(mask);
+    debug!(
+        "dirfd = {}, filename = {:?}, flags = {:?},mask = {:?},statx_buf_ptr = 0x{:x}",
+        dirfd, filename, flags, mask, statx_buf_ptr,
+    );
+
+    println!("filename = {:?}", filename);
+
+    if filename.is_empty() && !flags.contains(StatxFlags::AT_EMPTY_PATH) {
+        return_errno_with_message!(Errno::ENOENT, "path is empty");
+    }
+
+    if flags.contains(StatxFlags::AT_STATX_FORCE_SYNC)
+        && flags.contains(StatxFlags::AT_STATX_DONT_SYNC)
+    {
+        return_errno_with_message!(Errno::EINVAL, "invalid flags");
+    }
+
+    if mask.contains(StatxMask::STATX_RESERVED) {
+        return_errno_with_message!(
+            Errno::EINVAL,
+            "mask reserved for future struct statx expansion"
+        );
+    }
+
+    let dentry = {
+        let filename = filename.to_string_lossy();
+        let fs_path = FsPath::new(dirfd, filename.as_ref())?;
+        let fs = ctx.posix_thread.fs().resolver().read();
+        if flags.contains(StatxFlags::AT_SYMLINK_NOFOLLOW) {
+            fs.lookup_no_follow(&fs_path)?
+        } else {
+            fs.lookup(&fs_path)?
+        }
+    };
```

Comment:
I noticed that both this function and `sys_fstatat` share common logic. Perhaps we can extract the common logic for better code maintainability, and to adhere to the principle "Don't repeat yourself".